internal/policies/apparmor/apparmor_test.go

```diff
@@ -0,0 +1,223 @@
+		"error on unloading profiles failing":            {destAlreadyExists: "machine", apparmorParserError: true, wantErr: true},
+		"error on save assets dumping failing":           {entries: defaultMachineProfile, noParserOutput: true, saveAssetsError: true, wantErr: true},
+		"error on profile being a directory":             {entries: []entry.Entry{{Key: "apparmor-machine", Value: "nested/"}}, noParserOutput: true, wantErr: true},
+		"error on absent profile":                        {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.nonexistent"}}, noParserOutput: true, wantErr: true},
+		"error on file as a directory":                   {entries: []entry.Entry{{Key: "apparmor-machine", Value: "usr.bin.foo/notadir"}}, noParserOutput: true, wantErr: true},
+		"error on read-only root directory, no entries":  {readOnlyApparmorDir: ".", noParserOutput: true, wantErr: true},
+		"error on read-only root directory with entries": {entries: defaultMachineProfile, readOnlyApparmorDir: ".", noParserOutput: true, wantErr: true},
+		"error on read-only machine directory":           {entries: defaultMachineProfile, destAlreadyExists: "machine", readOnlyApparmorDir: "machine", noParserOutput: true, wantErr: true},
+		"error on read-only .old directory":              {entries: defaultMachineProfile, destAlreadyExists: "machine.old", readOnlyApparmorDir: "machine.old", noParserOutput: true, wantErr: true},
+		"error on read-only .new directory":              {entries: defaultMachineProfile, destAlreadyExists: "machine.new", readOnlyApparmorDir: "machine.new", noParserOutput: true, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		name := name
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			apparmorDir := t.TempDir()
+			parserCmdOutputFile := filepath.Join(t.TempDir(), "parser-output")
+			apparmorParserCmd := mockApparmorParserCmd(t, parserCmdOutputFile)
+			if tc.noApparmorParser {
+				apparmorParserCmd = []string{"this-definitely-does-not-exist"}
+			}
+			if tc.apparmorParserError {
+				apparmorParserCmd = append(apparmorParserCmd, "-Exit1-")
+			}
+
+			object := "machine"
+			if tc.user {
+				object = "users"
+			}
+
+			if tc.destAlreadyExists != "" {
+				require.NoError(t, os.RemoveAll(apparmorDir), "Setup: can't remove apparmor dir before filing it")
+				require.NoError(t,
+					shutil.CopyTree(
+						filepath.Join("testdata", "ApplyPolicy", "apparmor_dir", object), filepath.Join(apparmorDir, tc.destAlreadyExists),
+						&shutil.CopyTreeOptions{Symlinks: true, CopyFunction: shutil.Copy}),
+					"Setup: can't create initial apparmor dir profiles content")
+				fmt.Println(filepath.Join(apparmorDir, tc.destAlreadyExists))
+			}
+			if tc.readOnlyApparmorDir != "" {
+				testutils.MakeReadOnly(t, filepath.Join(apparmorDir, tc.readOnlyApparmorDir))
+			}
+			sat := sat{err: tc.saveAssetsError}
```

Comment:
what does sat stands for? (I had to read it twice to understand that this is the asset dumper)

Reply:
I was sure this was on me :p But yeah, probably SaveAssetsTo, mind renaming (and so, same in scripts, even if it’s not really part of that PR but does it worth its own PR?) to "mockAssetDumper" or anything similar?

Reply:
if we extract it somewhere, it will be specific test helper for managers (it doesn’t make sense out of the current manager context I think?)